### PR TITLE
Merge code for Stack modules

### PR DESCRIPTION
**Pull request by invincibleJai**
Changes related to stack modules are made.

### CODE REVIEW

**Comment by pmuir** [comment]
@invincibleJai can you make sure you rebase this to remove the merge commits?

**Comment by pmuir** [comment]
Can you remove the cico file changes please - you just changed the permissions on them for some reason.

File: src/app/analyze/analyze-routing.module.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { RouterModule, Routes } from '@angular/router';
 import { AnalyzeComponent } from './analyze.component';
 import { AnalyzeOverviewComponent } from './analyze-overview/analyze-overview.component'
 
+import {RenderStackDetailsComponent} from './stackreports/render-stack-details/render-stack-details.component';
```
Comment (pmuir) [inline src/app/analyze/analyze-routing.module.ts]: You need to keep the stack details stuff as a module like I did. Please can you remove the changes from this file

File: src/app/analyze/analyze.module.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { NgModule }         from '@angular/core';
+import { NgModule, CUSTOM_ELEMENTS_SCHEMA }         from '@angular/core';
```
Comment (pmuir) [inline src/app/analyze/analyze.module.ts]: What do you need this for?

File: src/app/analyze/analyze.module.ts [hunk 3]
```diff
@@ -7,9 +7,15 @@ import { AnalyzeRoutingModule } from './analyze-routing.module';
 
 import { AnalyzeOverviewModule } from './analyze-overview/analyze-overview.module';
 
+import {RenderStackDetailsComponent} from './stackreports/render-stack-details/render-stack-details.component';
```
Comment (pmuir) [inline src/app/analyze/analyze.module.ts]: You don't need this import

File: src/app/analyze/analyze.module.ts [hunk 4]
```diff
@@ -7,9 +7,15 @@ import { AnalyzeRoutingModule } from './analyze-routing.module';
...
+import {StackModule} from './stackreports/render-stack-details/module';
+
+import {StackRecoModule} from './stackreports/stack-recommendation/module';
```
Comment (pmuir) [inline src/app/analyze/analyze.module.ts]: Don't use abbreviations in names. Copy the naming approach I did in the stack-details module for file names.

File: src/app/analyze/analyze.module.ts [hunk 5]
```diff
@@ -7,9 +7,15 @@ import { AnalyzeRoutingModule } from './analyze-routing.module';
 
 import { AnalyzeOverviewModule } from './analyze-overview/analyze-overview.module';
 
+import {RenderStackDetailsComponent} from './stackreports/render-stack-details/render-stack-details.component';
+import {StackModule} from './stackreports/render-stack-details/module';
```
Comment (pmuir) [inline src/app/analyze/analyze.module.ts]: My refactor changed the location of the StackDetails module, please can you not move it.

File: src/app/analyze/analyze.module.ts [hunk 6]
```diff
@@ -7,9 +7,15 @@ import { AnalyzeRoutingModule } from './analyze-routing.module';
...
+
+import {StackRecoModule} from './stackreports/stack-recommendation/module';
+
 @NgModule({
-  imports:      [ AnalyzeOverviewModule, CommonModule, AnalyzeRoutingModule, HttpModule ],
+  imports:      [ StackModule, AnalyzeOverviewModule, CommonModule, AnalyzeRoutingModule, HttpModule ],
   declarations: [ AnalyzeComponent ],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA]
```
Comment (pmuir) [inline src/app/analyze/analyze.module.ts]: Why do you need this?

File: src/app/analyze/stackreports/container-toggler/container-toggler.component.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import {Component, ViewEncapsulation, Output, EventEmitter} from '@angular/core';
```
Comment (pmuir) [inline src/app/analyze/stackreports/container-toggler/container-toggler.component.ts]: I removed the stackreports directory, please merge these changes in to the stack-details module

File: src/app/analyze/stackreports/container-toggler/container-toggler.component.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    prev() : void {
+        this.prevBtn.emit('P');
+    }
+
+    next() : void {
+        this.nextBtn.emit('N');
+    }
+}
```
Comment (pmuir) [inline src/app/analyze/stackreports/container-toggler/container-toggler.component.ts]: Add new line at end of file.

File: src/app/analyze/stackreports/container-toggler/container-toggler.html [hunk 9]
```diff
@@ -0,0 +1,4 @@
+<div class="container-toggler">
+    <span (click)="prev()" class="prev">Previous</span>
+    <span (click)="next()" class="next">Next</span>
+</div>
```
Comment (pmuir) [inline src/app/analyze/stackreports/container-toggler/container-toggler.html]: Add new line at end of file.

File: src/app/analyze/stackreports/container-toggler/container-toggler.component.ts (hunk 7 above)
Comment (pmuir) [inline src/app/analyze/stackreports/container-toggler/container-toggler.component.ts]: In other words, there should be no files under stackreports.

File: src/app/analyze/stackreports/container-toggler/module.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import {NgModule, CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
+import {CommonModule} from '@angular/common';
+
+import {ContainerTogglerComponent} from './container-toggler.component';
+
+@NgModule({
+    imports: [CommonModule],
+    declarations: [ContainerTogglerComponent],
+    schemas: [CUSTOM_ELEMENTS_SCHEMA],
```
Comment (pmuir) [inline src/app/analyze/stackreports/container-toggler/module.ts]: Why do you need CUSTOM_ELEMENTS_SCHEMA?

File: src/app/analyze/stackreports/container-toggler/module.ts [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+    exports: [ContainerTogglerComponent]
+})
+
+export class ContainerTogglerModule {}
```
Comment (pmuir) [inline src/app/analyze/stackreports/container-toggler/module.ts]: New line needed

File: src/app/analyze/stackreports/render-component.service.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+import { Http, Response } from '@angular/http';
+
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/map';
+
+import { StackAnalysesModel } from './stack-analyses.model';
+
+@Injectable()
+export class RenderComponentService {
```
Comment (pmuir) [inline src/app/analyze/stackreports/render-component.service.ts]: I renamed this, please check the stack directory.

**Comment by pmuir** [comment]
This has not been properly merged against my changes in master - we can work through it tomorrow.

**Comment by invincibleJai** [comment]
This merge request has become obsolete and new one is #109

**Pull request closed without merging**